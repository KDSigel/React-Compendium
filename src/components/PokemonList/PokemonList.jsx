import React from "react";
import Pokemon from "../Pokemon/Pokemon";

export default function PokemonList({ pokemonList }) {
  console.log(pokemonList)
  return (
    <div>
      <ul aria-label='list'>
        {pokemonList.map(
          ({ pokemon, hp, height, attack, defense, url_image }) => (
            <li key={pokemon}>
              <Pokemon
                pokemon={pokemon}
                hp={hp}
                height={height}
                attack={attack}
                defense={defense}
                image={url_image}
              />
            </li>
          )
        )}
      </ul>
    </div>
  );
}
