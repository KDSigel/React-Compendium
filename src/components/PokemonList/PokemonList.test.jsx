import { screen, render } from "@testing-library/react";
import PokemonList from "./PokemonList";

it('testing my lists', () => {
    const fakePokes = [{
        'pokemon': 'butterfree',
        'hp': '60',
        'height': '11',
        'attack': '45',
        'defense': '50',
        'image':'http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png'
    },
    {   'pokemon': 'venusaur-mega',
        'hp': '80',
        'height': '24',
        'attack': '100',
        'defense': '123',
        'image':'http://assets.pokemon.com/assets/cms2/img/pokedex/full/003_f2.png'
    }
]
    render(<PokemonList pokemonList={fakePokes}/>)
    const pokelist = screen.getByLabelText('list')
    expect(pokelist).toMatchSnapshot()
}

)