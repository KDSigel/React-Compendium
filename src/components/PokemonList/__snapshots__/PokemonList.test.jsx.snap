// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing my lists 1`] = `
<ul
  aria-label="list"
>
  <li>
    <div>
      <img
        alt="poke"
      />
      <p>
        Name: 
        butterfree
      </p>
      <p>
        HP: 
        60
      </p>
      <p>
        Height: 
        11
      </p>
      <p>
        Attack: 
        45
      </p>
      <p>
        Defense: 
        50
      </p>
    </div>
  </li>
  <li>
    <div>
      <img
        alt="poke"
      />
      <p>
        Name: 
        venusaur-mega
      </p>
      <p>
        HP: 
        80
      </p>
      <p>
        Height: 
        24
      </p>
      <p>
        Attack: 
        100
      </p>
      <p>
        Defense: 
        123
      </p>
    </div>
  </li>
</ul>
`;
